### DISCUSSION TITLE
File name behavior not working, uploading same file twice

### DISCUSSION THREAD

**Discussion by StevenKnoop** [serenity-is/Serenity#6270] [Help]
I currently have to update a file entry twice for the file to be successfully updated. When debugging I find that on the first attempt the file is deleted, on the second attempt the file is successfully uploaded.

The desired result is to upload the file once and it should update successfully. (No file name suffixes or missing files). 

This is the current FileUploadEditor attribute settings:
`[FileUploadEditor(FilenameFormat = "Reports/{4}", AllowNonImage = true, CopyToHistory = true, OriginalNameProperty = "FileName")]`

I want to do some custom code to overwrite this, I implemented a new custom FileUploadEditor and created overriding OnBeforeSave and OnAfterSave methods. However, these methods only fire after the default Serenity FileUploadEditor has fired and completed their OnBeforeSave and OnAfterSave methods.

Your help would be greatly appreciated.

**Comment by edwardch** [comment]
I would also like to override the default ImageUploadBehavior / MultipleImageUploadBehavior code.

Has anyone else got anything they could point @StevenKnoop or myself to?

TIA!
